server/container_list.go

```diff
@@ -28,6 +28,7 @@ func filterContainer(c *pb.Container, filter *pb.ContainerFilter) bool {
 
 // ListContainers lists all containers by filters.
 func (s *Server) ListContainers(ctx context.Context, req *pb.ListContainersRequest) (*pb.ListContainersResponse, error) {
+	s.Update()
```

Comment:
could you move each of this below the `logrus.Debugf` call, otherwise, it's impossible to understand which call is currently calling `s.Update`